### PR TITLE
feat: [NPM] Add config file and feature toggles

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: npm/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+func main() {
+	klog.Infof("Start NPM version: %s", version)
+	cobra.CheckErr(rootCmd.Execute())
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	rootCmd.Flags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.azure-npm-debug-cli.yaml)")
```
Comment (rbtr) [inline npm/cmd/main.go:58]: help string is misleading, maybe:
```suggestion
	rootCmd.Flags().StringVar(&cfgFile, "config", "", fmt.Sprintf("config file (default is %s)", npmconfig.GetConfigPath()))
```

File: npm/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "azure-npm",
+	Short: "Collection of functions related to Azure NPM's debugging tools",
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		config := &npmconfig.Config{}
+		err := viper.Unmarshal(config)
+		if err != nil {
+			fmt.Printf("unable to decode into config struct, %v", err)
```
Comment (rbtr) [inline npm/cmd/main.go:32]: since this is a `RunE`, probably better to wrap and return

File: npm/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		}
+
+		Start(*config)
```
Comment (rbtr) [inline npm/cmd/main.go:35]: similarly, this func should block or return an error, and we should `return Start(...)`

File: npm/cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/spf13/viper"
+	"k8s.io/klog"
+)
+
+// Version is populated by make during build.
+var version string
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
```
Comment (rbtr) [inline npm/cmd/main.go:24]: very subjective, but i prefer to put the root command in a `root.go`, and keep `main.go` very lean, with only a `main()` and the min cobra imports

File: npm/cmd/debugcmd.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package cmd
+package main
```
Comment (rbtr) [inline npm/cmd/debugcmd.go:1]: the package is cmd, isn't it? and we can drop the "cmd" from this filename.

File: npm/util/errors/errors.go [hunk 6]
```diff
@@ -1,12 +1,18 @@
...
 import (
+	"errors"
 	"fmt"
 	"strings"
 
 	"github.com/Azure/azure-container-networking/npm/util"
 )
 
+var (
+	SrcNotSpecified = errors.New("source not specified")
```
Comment (rbtr) [inline npm/util/errors/errors.go]: should these be in the package where they get used instead of a `util`?

**Review by rbtr — COMMENTED** [review]
few organizational nitpicks, ❤️ consolidating on cobra and a proper config pkg

**Review by vakalapa — APPROVED** [review]
Lgtm 🚀

**Pull request merged**